scripts/smoke_test.sh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash
+
+if [[ $1 == "" ]]; then
+  echo "Missing test image!"
+  exit 1
+fi
+
+image_name=$(basename $1)
+echo "=== Smoke testing ${image_name}"
+
+[[ $image_name =~ ^(.+)-(libvirt|ovirt)-([0-9]*)([a-z]*)(.+)\.qc2$ ]]
+product_name=${BASH_REMATCH[1]}              #manageiq
+platform=${BASH_REMATCH[2]}                  #libvirt
+product_version_number=${BASH_REMATCH[3]}    #18? (this is almost always blank, will fill in later)
+product_version_name=${BASH_REMATCH[4]}      #quinteros?
+product_version_remainder=${BASH_REMATCH[5]} #-1-20240403
+# Fill in product version number if missing
+if [[ $product_version_number == "" ]]; then
+  letters=({a..z})
+  letter=${product_version_name::1}
+  for (( i=0; ; i++ ))
+  do
+    if [[ ${letters[$i]} == $letter ]]; then
+      product_version_number=$(($i + 1))
+      break
+    fi
+  done
+fi
```

Comment:
Is this basically just getting the ordinal value of the first character?  If so, I think this is a lot simpler:

```suggestion
if [[ "$product_version_number" == "" ]]; then
 product_version_number=$(($(printf '%d' "'${product_version_name::1}") - 96))
fi
```

Reply:
```
$ product_version_name=quinteros
$ echo $(($(printf '%d' "'${product_version_name::1}") - 96))
17
```